### ISSUE TITLE
Add branch visibility, consolidate branch actions

### ISSUE THREAD

**Issue by weihwang** [jupyterlab/jupyterlab-git#244]
UX goals: Increase visibility to target branches, in preparation for push/pull actions. This proposal also addresses issue #224 by focusing on git repo information in the header. Consolidating actions to the local/remote branches into an update modal gives us a clear header, and one place to take action on branches. https://www.figma.com/file/UyNpeQNuTj6Uv74MoEJDh3ji/Git-extension?node-id=524%3A3110. Pre-requisite UX change for #234 

Existing UI
![image](https://user-images.githubusercontent.com/551333/46969126-e95b0500-d069-11e8-9038-a8f630b8bffa.png)

v1 + visibility to remote branch
![image](https://user-images.githubusercontent.com/551333/47044652-9eaebb00-d145-11e8-888f-da86520e25ca.png)

v2 + ability to update local/remote branches
![image](https://user-images.githubusercontent.com/551333/47044695-b7b76c00-d145-11e8-8c4a-7bc30c1c6599.png)

**Comment by telamonian** [comment]
What was this issue supposed to be about, and should we still be concerned about it? The mockups are hard to parse, and aside from the text alignment I can't tell the difference between the first two.

@tgeorgeux Can you please interpret?

**Comment by fcollonval** [comment]
@telamonian @kgryte I think this can be closed.

**Comment by kgryte** [comment]
Agreed.
